### DISCUSSION TITLE
Is there a cycleway here? => Mapped separately could benefit from highlighting cycleways in render-distance

### DISCUSSION THREAD

**Discussion by kmpoppe** [streetcomplete/StreetComplete#3912] [Ideas]
The "OSM Apps & Editoren" Telegram Group [discussed](https://t.me/osmappsundeditoren/3016) whether it would be beneficial to have StreetComplete highlight cycleways around the question pin "Is there a cycleway here?".

The issue leading to that discussion was how one would be able to guess whether a way on the map was a cycleway that would belong to the street in question.

Example:
<img src="https://user-images.githubusercontent.com/24451207/159788440-8052655a-deb7-4f7b-9cd4-210b2a5f165e.jpg" width="300">

[Konrad-Adenauer-Allee](https://www.openstreetmap.org/way/889371627) has it's cycleway [here](https://www.openstreetmap.org/way/788694797)

Highlighting the way just like buildings when the housenumber was tagged or fire hydrants or benches within render distance, this could like this (just a quick mspaint-mockup):
<img src="https://user-images.githubusercontent.com/24451207/159789323-6e89d38d-8a7b-4933-881f-3043cb3e724e.png" width="300">

Does this have any chance of an implementation? I wouldn't know how to solve this programatically.

Kai

**Comment by joshinils** [comment]
one way would be to check for separate cycleways or sidewalks without `is_sidepath=yes` or `is_sidepath:of=*` or `is_sidepath:of:name`.
and for streets with `foot=use_sidepath` without `sidewalk:*=separate` or `bicycle=use_sidepath|optional_sidepath` without `cycleway:*=separate` or vice versa.

and then make the connection between the street and separate way via the `is_sidepath:*` tags.

that way any street or sidepath without the correct set of tags could be searched for and corrected.
just like an address asks for the accompanying road to set the name, this can set the `is_sidepath:of*` tags and at the same time "fix" the road tags.

**Comment by westnordost** [comment]
Hmm, why would you assume that the path you marked pink in your mockup is not a cycleway that is displayed separately on the map? After all, you are on-site when answering the quest. Your blue dot is likely **on** that path when answering it.

**Reply by joshinils** [reply]
it could be tagged purely as a sidewalk (or worse a plain path), which i think would look the same

**Reply by kmpoppe** [reply]
I might be standing on the opposite side of the dual carriageway that has a tram line running through and thus I may either be unable to see what kind of way it is, or, what OP was talking about, I have no way of knowing whether the way is tagged as being a Cycleway at all (the answer implies that you already know it **is tagged as a CW** and that you don't because you can't see the tags in SC - for very good reasons)

**Comment by BalooUriza** [comment]
Sidewalk implies bicycle=no

**Reply by kmpoppe** [reply]
I, for one, am not able to spot any difference in a sidewalk or cycleway from the background map rendered. Again, without seeing the tagging of the way, I have _no_ way of knowing what it's tagged as.

**Reply by matkoniecz** [reply]
Also, "Sidewalk implies bicycle=no" is not generally true (for example in Poland it implies "no, with exceptions X, Y, Z")

**Comment by Helium314** [comment]
See #3594, which started #3603 and #3609.
https://github.com/streetcomplete/streetcomplete-mapstyle/issues/94 might also be interesting here.

What is clear from https://github.com/streetcomplete/StreetComplete/issues/3594#issuecomment-997012895 is that the whole bbox of the cycleway needs to be queried for getting highlighted elements, and this could be *really* huge.

SQLite with R\*tree should support querying not only by bbox, but also by other shapes (e.g. polygon created from cycleway + 20 m padding). I would expect this to be faster than querying a huge bbox, but didn't test it. 
In the end SC did not switch to an R\*tree compatible library because the performance was not significantly faster than optimizing the existing DB (https://github.com/streetcomplete/StreetComplete/issues/3609#issuecomment-1031177576 and below)
